dashboard/test/controllers/programming_expressions_controller_test.rb

```diff
@@ -139,9 +139,9 @@ class AccessTests < ActionController::TestCase
     test_user_gets_response_for :update, params: -> {@update_params}, user: :teacher, response: :forbidden
     test_user_gets_response_for :update, params: -> {@update_params}, user: :levelbuilder, response: :success
 
-    test_user_gets_response_for :show, params: -> {{id: @programming_expression.id}}, user: nil, response: :success
-    test_user_gets_response_for :show, params: -> {{id: @programming_expression.id}}, user: :student, response: :success
-    test_user_gets_response_for :show, params: -> {{id: @programming_expression.id}}, user: :teacher, response: :success
-    test_user_gets_response_for :show, params: -> {{id: @programming_expression.id}}, user: :levelbuilder, response: :success
+    test_user_gets_response_for :show, params: -> {{programming_environment_name: @programming_expression.programming_environment.name, programming_expression_key: @programming_expression.key}}, user: nil, response: :success
```

Comment:
It would be good add some tests to check that only levelbuilders can use `show_by_id`

Reply:
I was waffling a bit here because I think this comes down to not caring who can see `show_by_id`. But test can be documentation so I'll add some